sylt-compiler/src/typechecker.rs

```diff
@@ -1543,15 +1569,15 @@ impl TypeChecker {
             AssignableKind::Expression(_) => err_type_error!(
                 self,
                 span,
-                ctx,
                 TypeError::Assignability,
                 "Cannot assign to expressions"
             ),
         }
     }
 
-    fn add_constraint(&mut self, a: usize, constraint: Constraint) {
-        self.find_node_mut(a).constraints.insert(constraint);
+    fn add_constraint(&mut self, a: usize, span: Span, constraint: Constraint) {
+        // TODO(ed): Don't reinsert stuff?
```

Comment:
:shrug:

Reply:
Constraints that are added multiple times overwrite the span - should we keep the oldest span? Or is there some better heuristic for sources of constraints? They're all equivalent but there might be a better way to do all of this? Or maybe the newest one is the most relevant?

But we now don't overwrite things.